zappa/core.py

```diff
@@ -1247,6 +1247,10 @@ def create_lambda_function(
                 ReservedConcurrentExecutions=concurrency,
             )
 
+        # Wait for lambda to become active, otherwise many operations will fail
+        # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/lambda.html#waiters
+        self.wait_until_lambda_function_is_ready(resource_arn)
+
         return resource_arn
 
     def update_lambda_function(
```

Comment:
Should we also add wait until the function is ready when we update the lambda function?

Reply:
I _think_ the function will always be in Active state after the initial deployment, so this would effectively be a no-op.

Reply:
kk good to leave as is then. easy to add in the future if we realize it's needed

Reply:
Waiting for the function to be ready is needed after updating the function as well - this is captured in issues such as #1035

Reply:
@javulticat I thought you were dealing with that in #1042 but I will make the changes to this PR if you'd prefer.

Reply:
Happy to consolidate in this PR if you don't mind 🙂

Reply:
Happy to consolidate in this PR if you don't mind 🙂

Reply:
@javulticat done, this PR is now checking lambda state prior to updating too.